driver-core/src/main/com/mongodb/connection/SocketSettings.java

```diff
@@ -62,8 +64,8 @@ public static Builder builder(final SocketSettings socketSettings) {
      * A builder for an instance of {@code SocketSettings}.
      */
     public static final class Builder {
-        private long connectTimeoutMS = 10000;
-        private long readTimeoutMS;
+        private int connectTimeoutMS = 10000;
+        private int readTimeoutMS;
```

Comment:
There is no reason to have the fields internally of the `long` type, because the millisecond values were not allowed (and will continue being not allowed) to be greater than `Integer.MAX_VALUE`.